devtools/pip-tools/pr_tool/backport.py

```diff
@@ -0,0 +1,317 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2023-2023 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+"""
+this module provides commands to do a backport of a specific
+bareos PR into another branch
+"""
+
+import logging
+from os import environ
+from sys import stdout, stderr
+from subprocess import run
+from tempfile import NamedTemporaryFile
+from importlib import resources
+from git import RemoteReference
+from git.exc import GitCommandError
+from .github import Gh, InvokationError, whoami
+
+
+class UnmanagedBranchException(Exception):
+    pass
+
+
+def git_editor(repo, file):
+    if "GIT_EDITOR" in environ:
+        editor = environ["GIT_EDITOR"]
+    else:
+        editor = repo.git.var("GIT_EDITOR")
+    res = run([editor, file], check=False)
+    return res.returncode == 0
+
+
+def find_user_remote(repo, username):
+    expected_url = f"git@github.com:{username}/bareos.git"
+    for remote in repo.remotes:
+        if expected_url == list(remote.urls)[0]:
```

Comment:
You only have multiple urls for a remote if you want to pull from or push to multiple locations at once.

So to make this bullet-proof we would have to look at all urls in the remote and make sure there is only one. Leaving us with something like
```python
remote_urls = list(remote.urls)
if len(remote_urls) > 1: continue
if expected_url == remote_urls[0]:
  ...
```

However, it may be a good idea to check that to make sure we don't automatically push to a weirdly configured remote.

Reply:
makes sense!